### DISCUSSION TITLE
Alternative for Dataproviders

### DISCUSSION THREAD

**Discussion by bajipogadadanda** [webdriverio/webdriverio#12053] [General]
Hi,
WDIO/Mocha doesn't support dataproviders, of-course we can use for-Each instead.
Since we have many existing tests with Jasmine data providers, I wanted to check if there is any alternative for Jasmine data providers in WDIO(if we use Mocha). shall we use npm packages like mocha-param? will it work? or any similar efficient libraries to replace Jasmine data providers to migrate my code to WDIO?
Pls let me know your thoughts on it.

**Comment by christian-bromann** [comment]
I am not aware of a similar alternative. I have the following questions:

- why don't you use WebdriverIO testrunner with Jasmine, would the data provider plugin work?
- _of-course we can use for-Each instead._ why don't you want to do it?

**Reply by bajipogadadanda** [reply]
we have chosen mocha as it has greater advantages than Jasmine.
Currently, we are using for-Each, however, we have many existing tests with protractor/jasmine data-providers. So, migrating all these tests to wdio with for-Each and eliminating data-providers (manually changing the code ) is tedious. So, trying to find an easy path for it.

**Reply by christian-bromann** [reply]
> So, migrating all these tests to wdio with for-Each and eliminating data-providers (manually changing the code ) is tedious.

We have a [codemod](https://github.com/webdriverio/codemod) project that allows you to modify this code at a higher scale. If you raise a PR there for this use case, I am happy to merge and release it.

**Reply by mathmul** [reply]
there is also [mocha-each](https://www.npmjs.com/package/mocha-each/v/1.0.1) you can get [inspiration](https://github.com/ryym/mocha-each/blob/master/lib/index.js) from (since it's pretty old).

**Comment by bajipogadadanda** [comment]
sure
